### PR TITLE
feat: rework OS exporter tests to support AWS

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

Adopting OpenSearch exporter module to run against AWS.

Note: there will be an abstraction layer introduced since there's a lot of copy-paste logic currently in this PR.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Related to: https://github.com/camunda/team-data-engine-maintenance/issues/49

### CODE REVIEW

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java [hunk 1]
```diff
@@ -14,83 +14,52 @@
...
-    CloseHelper.quietCloseAll(testClient, client);
+  private static SearchDBExtension initializeSearchDBExtension() {
+    final var openSearchAwsInstanceUrl =
+        Optional.ofNullable(System.getProperty(IT_OPENSEARCH_AWS_INSTANCE_URL_PROPERTY)).orElse("");
+    if (openSearchAwsInstanceUrl.isEmpty()) {
+      return new ContainerizedSearchDBExtension();
+    } else {
+      return new AWSSearchDBExtension(openSearchAwsInstanceUrl);
+    }
   }
```
Comment (ChrisKujawa) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java:43]: 🔧 I think I would make this factory method part of the SearchDBextension. 

Then like:

```
@RegisterExtension private static SearchDBExtension searchDB = SearchDBExtension.create();
```

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java [hunk 2]
```diff
@@ -152,24 +127,24 @@ void shouldPutComponentTemplate() {
...
-            RestClientFactory.of(config, true),
-            indexRouter,
-            templateReader,
+            searchDB.config(),
+            searchDB.bulkRequest(),
+            RestClientFactory.of(searchDB.config(), true),
+            searchDB.indexRouter(),
+            searchDB.templateReader(),
             new OpensearchMetrics(new SimpleMeterRegistry()));
     authenticatedClient.putComponentTemplate();
```
Comment (ChrisKujawa) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java:144]: 🔧 Maybe we want in the API actually a method to recreate the client with certain configs? Then we can get rid of the other properties exposed in the interface? 🤔

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java (hunk 1 above)
Comment (igpetrov) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java:43]: Fair point

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java (hunk 2 above)
Comment (igpetrov) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java:144]: If I understood your question correctly, we cannot do this. All those API methods `recordFactory()`, `indexRouter()`, etc methods are used in tests.

**Comment by igpetrov** [comment]
Changes moved here: https://github.com/camunda/camunda/pull/24986 as we wanted to push first pipeline together with tests to make sure both work in pair

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java (hunk 1 above)
Comment (igpetrov) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java]: Changes moved here https://github.com/camunda/camunda/pull/24986

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java (hunk 2 above)
Comment (igpetrov) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java:135]: PR moved here: https://github.com/camunda/camunda/pull/24986

**Pull request closed without merging**